packages/hash/api/src/graphql/typeDefs/knowledge/page.typedef.ts

```diff
@@ -0,0 +1,232 @@
+    """
+    The JSON object containing the entity's properties.
+    """
+    properties: JSONObject!
+    # ENTITY INTERFACE FIELDS END #
+  }
+
+  type KnowledgeEntityRef {
+    accountId: ID!
+    entityId: ID!
+    entityVersion: String!
+  }
+
+  """
+  Insert a block into a page with a corresponding entity.
+  """
+  input KnowledgeInsertBlockAction {
+    """
+    The account ID to create the block and associated entity in.
+    """
+    accountId: ID!
+    """
+    The position in the page to place the block.
+    """
+    position: Int!
+    """
+    The block componentId.
+    """
+    componentId: ID
+    """
+    The block entity to insert into the page. You should not set a componentId
+    if you provide this
+    """
+    existingBlockEntity: ExistingEntity
+    """
+    The entity to associate with the new block
+    """
+    entity: KnowledgeEntityDefinition!
+    """
+    Allows UpdatePageContentsActions to reference entities created in other actions. Also allows callers to updatePageContents to find the entity id created for this definition in the result. See UpdatePageContentsResult.
+    """
+    blockPlaceholderId: ID
+    """
+    Allows UpdatePageContentsActions to reference entities created in other actions. Also allows callers to updatePageContents to find the entity id created for this definition in the result. See UpdatePageContentsResult.
+    """
+    entityPlaceholderId: ID
+  }
+
+  """
+  Remove a block from a page.
+  """
+  input KnowledgeRemoveBlockAction {
+    """
+    The position of the block to remove from the page.
+    """
+    position: Int!
+  }
+
+  """
+  Move a block within a page.
+  """
+  input MoveBlockAction {
+    """
+    The current position of the block.
+    """
+    currentPosition: Int!
+    """
+    The position to move the block to.
+    """
+    newPosition: Int!
+  }
+
+  """
+  Update an entity in a page.
+  """
+  input KnowledgeUpdateEntityAction {
+    """
+    The account the entity resides in.
+    """
+    accountId: ID!
+    """
+    The entity's fixed ID.
+    """
+    entityId: ID!
+    """
+    The entity's new properties.
+    """
+    properties: JSONObject!
+  }
+
+  """
+  Swap a blocks data
+  """
+  input KnowledgeSwapBlockDataAction {
+    """
+    The account the block resides in
+    """
+    accountId: ID!
+
+    """
+    The Block entity's fixed ID
+    """
+    entityId: ID!
+
+    """
+    The account the new entity resides in
+    """
+    newEntityAccountId: ID!
+
+    """
+    The new entity's fixed ID
+    """
+    newEntityEntityId: ID!
+  }
+
+  """
+  Create an entity, which you can then reference in other actions, such as a InsertBlockAction
+  """
+  input KnowledgeCreateEntityAction {
+    entity: KnowledgeEntityDefinition!
+    entityPlaceholderId: ID
+    accountId: ID!
+  }
+
+  """
+  Create an entity type, which you can then reference in future CreateEntityActions
+  """
+  input KnowledgeCreateEntityTypeAction {
```

Comment:
```suggestion
  input CreateKnowledgeEntityTypeAction {
```

Reply:
Addressed in 021f42e97909338ec507cf74b5757ed426586e6d